docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+# Create Smart Contracts Basics
+
+A smart contract is a program that runs on the blockchain. It allows you to add functionality to an account ranging from simple ones like a todo application to a fully-fledged RPG game which runs entirely on the blockchain.
+
+This guide will show you how to develope a basic EOS smart contract with **DUNE** and with the C++ programming language. Also the basic smart contract concepts will be explained along the way.
+
+## Create Contract and Test Accounts
+
+To deploy a smart contract you need an account to deploy it to. Create an account `hello` with the following command:
+
+```shell
+dune --create-account hello
+```
+
+Create a second account, `ama`, for test purposes.
+
+```shell
+dune --create-account ama
+```
+
+## Create a DUNE Smart Contract Application
+
+Make sure you have DUNE installed. Otherwise follow the [DUNE development setup](../10_getting-started/10_dune-guide/index.md#installation) documentation.
+
+There are two types of DUNE smart contract applications.
+
+- [The bare application](#create-build-and-deploy-a-dune-bare-application)
```

Comment:
Is there any reason to document both?

It seems like cmake has better DUNE support (since bare doesn't even have a dedicated build command)

Reply:
one is simple because it has much less files but a more complicated build command.
the other (cmake) has a simple build command but it is more complicated because the cmake project is more compex.

if I had to chose I would go with the cmake one but I can easily see other devs pick up the first one cause it is much simpler from the files organization point of view.

Reply:
Alright, let's keep it this way. I'll add a note to see if we can make compilation simpler for CDT in the future.

Reply:
we discussed this in our meeting and we agreed to stick with the cmake version cause it is simpler for the user.